### PR TITLE
chore: calculate CPU quota allocation

### PR DESCRIPTION
**Pull request by ascorbic**
When run in a container, neither physical cores not logical threads are accurate representations of the available CPU. With this PR we use data from the scheduler to find out what we actually have allocated, then work out the number of physical cores available based on that.

### CODE REVIEW

File: plugin/src/cli.ts [hunk 1]
```diff
@@ -63,9 +101,20 @@ async function run() {
   // This value has been tested and gives best peformance for most scenarios.
   if (!GATSBY_CPU_COUNT && process.env.NETLIFY && !process.env.NETLIFY_LOCAL) {
     const reportedCores = cpuCoreCount(true)
-    GATSBY_CPU_COUNT = Math.max(reportedCores - 2, 2).toString()
+    const threads = os.cpus().length
+    const threadsPerCore = threads / reportedCores
+
+    let cpuAllocation = getCpuAllocation()
+    cpuAllocation ||= threads
```
Comment (ericapisani) [inline plugin/src/cli.ts:108]: [dust] I think this could also be written as 
```suggestion
    const cpuAllocation = getCpuAllocation() || threads
```

**Review by ericapisani — APPROVED** [review]
LGTM

File: plugin/src/cli.ts [hunk 2]
```diff
@@ -189,8 +236,10 @@ async function run() {
 
     console.log(
       stripIndent`
-      Built site using the experimental ${green`Netlify Gatsby build runner`}. 
+
+      🏃 Built site using the experimental ${green`Netlify Gatsby build runner`}
```
Comment (nickytonline) [inline plugin/src/cli.ts:240]: I would not have approved this PR if the runner emoji was missing. 😎

**Review by nickytonline — APPROVED** [review]
🏃🏻‍♂️

File: plugin/src/cli.ts (hunk 1 above)
Comment (ascorbic) [reply plugin/src/cli.ts:108]: Haha. Thanks. Not sure what I was thinking there.

**Review by ericapisani — APPROVED** [review]

**Pull request merged**